src/physics/Body.js

```diff
@@ -540,22 +544,31 @@ Phaser.Plugin.Isometric.Body.prototype = {
 
         this.phase = 2;
 
-        if (this.deltaX() < 0) {
-            this.facing = Phaser.Plugin.Isometric.BACKWARDX;
-        } else if (this.deltaX() > 0) {
-            this.facing = Phaser.Plugin.Isometric.FORWARDX;
-        }
-
-        if (this.deltaY() < 0) {
-            this.facing = Phaser.Plugin.Isometric.BACKWARDY;
-        } else if (this.deltaY() > 0) {
-            this.facing = Phaser.Plugin.Isometric.FORWARDY;
+        // stops sprites flying off if isoPosition is changed during update
+        if (this._reset) {
+            this.prev.x = this.position.x;
+            this.prev.y = this.position.y;
+            this.prev.z = this.position.z;
         }
 
-        if (this.deltaZ() < 0) {
-            this.facing = Phaser.Plugin.Isometric.DOWN;
-        } else if (this.deltaZ() > 0) {
-            this.facing = Phaser.Plugin.Isometric.UP;
+        if (this.deltaAbsX() >= this.deltaAbsY() && this.deltaAbsX() >= this.deltaAbsZ()){
+            if (this.deltaX() < 0) {
+                this.facing = Phaser.Plugin.Isometric.BACKWARDX;
+            } else if (this.deltaX() > 0) {
+                this.facing = Phaser.Plugin.Isometric.FORWARDX;
+            }
+        } else if (this.deltaAbsY() >= this.deltaAbsX() && this.deltaAbsY() >= this.deltaAbsZ()){
+            if (this.deltaY() < 0) {
+                this.facing = Phaser.Plugin.Isometric.BACKWARDY;
+            } else if (this.deltaY() > 0) {
+                this.facing = Phaser.Plugin.Isometric.FORWARDY;
+            }
+        } else {
+            if (this.deltaZ() < 0) {
+                this.facing = Phaser.Plugin.Isometric.DOWN;
+            } else if (this.deltaZ() > 0) {
+                this.facing = Phaser.Plugin.Isometric.UP;
+            }
```

Comment:
Previously, if you moves in X, Y and Z directions, it would always be facing up or down. Even if it was (100,200,1), now it looks as you would expect visually. (I used facing to change texture, example here: http://wassname.org/grumpyCat)